File: Question4.cs
using System;
namespace Algorithm_Practice {
    class Question4 {
        public Question4() {
        }

        public void Action(int n, int m) {
            if (m > n) {
                int tmp = m;
                m = n;
                n = tmp;
            }

            Console.WriteLine("GCD({0:d},{1:d}) = {2:d}", n, m, CalcGCD(n, m));
        }

        private int CalcGCD(int n, int m) {
            if (n == 0) {
                return m;
            } else {
                return CalcGCD(m % n, n);
            }
        }
    }
}

File: Question4.java
class Question4 {

	public static void main(String[] args) {
		// TODO 自動生成されたメソッド・スタブ
		int n = 1029;
		int m = 1071;
		System.out.printf("GCD(%d, %d) = %d\n", n, m, calcGCD(n, m));
	}

	static int calcGCD(int n, int m) {
		if (n == 0) {
			return m;
		} else {
			return calcGCD(m % n, n);
		}
	}
}
